src/user_config.rs

```diff
@@ -0,0 +1,236 @@
+use dirs;
+use failure::err_msg;
+use serde::{Deserialize, Serialize};
+use std::fs;
+use std::path::{Path, PathBuf};
+use termion::event::Key;
+
+const FILE_NAME: &str = "config.yml";
+const CONFIG_DIR: &str = ".config";
+const APP_CONFIG_DIR: &str = "spotify-tui";
+
+fn parse_key(key: String) -> Result<Key, failure::Error> {
+    fn get_single_char(string: &str) -> char {
+        match string.chars().nth(0) {
+            Some(c) => c,
+            None => panic!(),
+        }
+    }
+
+    match key.len() {
+        1 => Ok(Key::Char(get_single_char(key.as_str()))),
+        _ => {
+            let sections: Vec<&str> = key.split('-').collect();
+
+            if sections.len() > 2 {
+                return Err(failure::format_err!(
+                    "Shortcut can only have 2 keys, \"{}\" has {}",
+                    key,
+                    sections.len()
+                ));
+            }
+
+            match sections[0].to_lowercase().as_str() {
+                "ctrl" => Ok(Key::Ctrl(get_single_char(sections[1]))),
+                "alt" => Ok(Key::Alt(get_single_char(sections[1]))),
+                "left" => Ok(Key::Left),
+                "right" => Ok(Key::Right),
+                "up" => Ok(Key::Up),
+                "down" => Ok(Key::Down),
+                "backspace" | "delete" => Ok(Key::Backspace),
+                "del" => Ok(Key::Delete),
+                "esc" | "escape" => Ok(Key::Esc),
+                "pageup" => Ok(Key::PageUp),
+                "pagedown" => Ok(Key::PageDown),
+                "space" => Ok(Key::Char(' ')),
+                _ => Err(failure::format_err!(
+                    "The key \"{}\" is unknown.",
+                    sections[0]
+                )),
+            }
+        }
+    }
+}
+
+fn check_reserved_keys(key: Key) -> Result<(), failure::Error> {
+    let reserved = [
```

Comment:
We can add more keys to check here, but if we want to add checks for keys that the user can re-bind, we would need to use a vec or maybe a hashmap to be able to not block the user from doing something like this:
```yaml
decrease_volume: '+'
increase_volume: '-'
```
Because with current code, if we add `+` and `-` to the reserved keys, the User wouldn't be able to do something like this, even if it is valid.

I think it isn't too important to block keys that are configurable from being reused by the user at all, as those keys are documented in the README.

Thoughts?